lib/util/cache/repository/impl/client/local/index.ts

```diff
@@ -0,0 +1,36 @@
+import upath from 'upath';
+import { GlobalConfig } from '../../../../../../config/global';
+import { logger } from '../../../../../../logger';
+import { outputCacheFile, readCacheFile } from '../../../../../fs';
+import type { CacheClient, RepoCacheRecord } from '../../../types';
+
+export class LocalRepoCache implements CacheClient {
+  constructor(private platform: string, private repository: string) {}
+
+  public getCacheFileName(): string {
+    const cacheDir = GlobalConfig.get('cacheDir');
+    const repoCachePath = '/renovate/repository/';
+    const platform = this.platform;
+    const fileName = `${this.repository}.json`;
+    return upath.join(cacheDir, repoCachePath, platform, fileName);
+  }
+
+  async read(): Promise<string | undefined> {
+    const cacheFileName = this.getCacheFileName();
```

Comment:
```suggestion
```

Reply:
this was originally present before this pr, i guess this have something to do with exceptions and the error message omitted in the catch block. `cacheFileName` is used there. 
this was basically extracted as is from the original code

Reply:
remove inner duplicate so the cache file name is in scope in the catch block